Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -543,6 +545,8 @@ - (void)hideInputIfNeeded
 
 - (void)viewDidLoad
 {
+    self.collectionView.layoutDelegate = self;
```

Comment:
nit: Rather than further diverging JSQMVC with this delegate protocol only to call back to delegate methods implemented in our own class, it seems to me like it would be simpler to skip all that and just override `setFrame` and `setBounds` in our own class.

Is there a reason that you did it this way?

Reply:
Yes, we need to override `setFrame` and `setBounds` - but they hang on `JSQMessagesCollectionView`, not `JSQMessagesViewController`, so we can't override them in `MessagesViewController`.  We're not using a custom subclass of `JSQMessagesCollectionView`.  I suppose we could, but it'd be a bunch of work and we're already diverging from JSQ, and JSQ is defunct, so...

Reply:
Whoops, my mistake. Thanks for the clarification.